src/python/ensembl/io/genomio/events_loader.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+from typing import Dict, List
+from dataclasses import dataclass
+
+import argschema
+from sqlalchemy.orm import Session
+
+from ensembl.database import DBConnection
+from ensembl.core.models import MappingSession, StableIdEvent
+
+
+@dataclass
+class IdEvent:
+    """Simple representation for the events from the input file"""
+    from_id: str
+    to_id: str
+    release: str
+    release_date: str
+
+
+class MapSession:
+    """Simple mapping_sessions representation from the input file"""
+    def __init__(self, release: str, release_date: str) -> None:
+        self.release = release
+        self.release_date = release_date
+        self.events: List[IdEvent] = []
+
+    def add_event(self, event: IdEvent) -> None:
+        """Add an event to this mapping_session"""
+        self.events.append(event)
+
+
+def load_events(input_file: Path) -> List[IdEvent]:
+    """Load events from input file.
+    Expected tab file columns: old_id, new_id, event_name, release, release_date
+    
+    """
+    events: List[IdEvent] = []
+
+    with input_file.open("r") as events_fh:
```

Comment:
I believe it is better to extend this method:
```suggestion
def load_events(input_file: PathLike) -> List[IdEvent]:
    """Load events from input file.
    Expected tab file columns: old_id, new_id, event_name, release, release_date
    
    """
    events: List[IdEvent] = []

    with Path(input_file).open("r") as events_fh:
```
_Note:_ you will need to add `from os import PathLike` above.